rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -513,7 +575,18 @@ void deserialize_array<std::string>(
 {
     (void)call_new;
     if (member->array_size_ && !member->is_upper_bound_) {
-        deser.deserializeArray(((rosidl_generator_c__String*)field)->data, member->array_size_);
+        auto deser_field = (rosidl_generator_c__String*)field;
+        std::string stringseq[member->array_size_];
+        deser.deserializeArray(stringseq, member->array_size_);
+        for (size_t i = 0; i < member->array_size_; ++i) {
+          // NOTE: Do we need to call init ?
```

Comment:
This should be answered / addressed before merging. On a first look I am not sure what the answer is though.